thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/v2/RootCauseMetricResource.java

```diff
@@ -588,13 +539,33 @@ private MetricSlice alignSlice(MetricSlice slice) throws Exception {
       granularity = slice.getGranularity();
     }
 
+    long offset = DateTimeZone.forID(timezone).getOffset(slice.getStart());
     long timeGranularity = granularity.toMillis();
-    long start = (slice.getStart() / timeGranularity) * timeGranularity;
-    long end = ((slice.getEnd() + timeGranularity - 1) / timeGranularity) * timeGranularity;
+    long start = ((slice.getStart() + offset) / timeGranularity) * timeGranularity - offset;
```

Comment:
Could you explain why we need to add and reduce an offset here?

Reply:
Time buckets are computed in days, but may need to be offset against UTC for clients in non-utc time zones, i.e. the bucket-boundaries should align to UTC + offset day boundaries instead of plain UTC day boundaries

Reply:
Could you leave a comment on the code segment? It is easier for people to understand it later.

Reply:
I added javadoc above

Reply:
I mean if you can leave a comment for line 542 to line 545. It is easier for people to quickly understand.